coursemapper-kg/recommendation/app/services/course_materials/recommendation/youtube_service.py

```
@@ -42,17 +43,65 @@
+
+                if retry_count == 0:
+                    return None, None
+            except HttpError as e:
+                if e.resp.status == 403 and "quota" in str(e):
+                    print(f"Quota exceeded for key: {key}. Trying next key...")
+                else:
+                    raise e
+        raise Exception("All API keys have exceeded their quota.")
+
 
     def get_videos(self, concepts, top_n=15):
```

Ruff (0.12.2)

Missing return type annotation for public function `get_videos`

(ANN201)